bindings/c/fdb_c.cpp

```diff
@@ -652,7 +652,70 @@ FDBFuture* fdb_transaction_get_mapped_range_impl(FDBTransaction* tr,
 }
 
 // TODO: Support FDB_API_ADDED in generate_asm.py and then this can be replaced with fdb_api_ptr_unimpl.
-FDBFuture* fdb_transaction_get_mapped_range_v699(FDBTransaction* tr,
+FDBFuture* fdb_transaction_get_range_and_flat_map_v699(FDBTransaction* tr,
+                                                       uint8_t const* begin_key_name,
+                                                       int begin_key_name_length,
+                                                       fdb_bool_t begin_or_equal,
+                                                       int begin_offset,
+                                                       uint8_t const* end_key_name,
+                                                       int end_key_name_length,
+                                                       fdb_bool_t end_or_equal,
+                                                       int end_offset,
+                                                       uint8_t const* mapper_name,
+                                                       int mapper_name_length,
+                                                       int limit,
+                                                       int target_bytes,
+                                                       FDBStreamingMode mode,
+                                                       int iteration,
+                                                       fdb_bool_t snapshot,
+                                                       fdb_bool_t reverse) {
+	fprintf(stderr, "UNIMPLEMENTED FDB API FUNCTION\n");
```

Comment:
If we completely remove this function, this can be a problem for upgrade. Imagine an application using this function needs to be upgraded from 7.0.0 to 7.1.0. For that the application will be started with client libraries for 7.0 and 7.1 and FDB_API_VERSION=700. It won't be possible to have a version of an application that is working both with 7.0 and 7.1, because in the 7.0 library, the new function is not available and in 7.1 library the old function is not available. If you want to remove this function, you can remove it only for 7.1.0 i.e. for (#if FDB_API_VERSION >= 710). Also see how other functions were remove in the past. This means that if the 7.1. client library is used in 7.0 compatibility mode (i.e. initialized with fdb_select_api_version(700)), it must find the function providing the same signature and behavior. You have a choice between preservinv the old implementation as it was or reimplement as it as adapter of the new function. In any case, we have to be careful about this, because the CI does not test the upgrade scenarios.

Reply:
Discussed offline. `fdb_transaction_get_range_and_flat_map` was added prematurely in 7.0, but nobody actually uses and nobody needs it, so we decide to just remove it.